### PR TITLE
Added CI based on github action

### PR DESCRIPTION
**Pull request by Pask423**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -18,7 +18,8 @@ lazy val commonSettings = commonSmlBuildSettings ++ ossPublishSettings ++ Seq(
   parallelExecution := false,
   // Sonatype OSS deployment
   credentials += Credentials(Path.userHome / ".ivy2" / ".credentials"),
-  Test / publishArtifact := false
+  Test / publishArtifact := false,
+  mimaPreviousArtifacts := Set.empty // we only use MiMa for `core` for now, using enableMimaSettings
```
Comment (adamw) [inline build.sbt:22]: there's no core module here, I think mima is misconfigured

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Before submitting pull request:
+- [ ] Check if the project compiles by running `sbt compile`
+- [ ] Verify docs compilation by running `sbt compileDocs`
```
Comment (adamw) [inline .github/pull_request_template.md:3]: are there any docs?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            ~/.ivy2/cache
+            ~/.coursier
+          key: sbt-cache-${{ runner.os }}-${{ hashFiles('project/build.properties') }}
+      - name: Compile
+        run:
+          sbt -v compile
+      - name: Test
+        run:
+          sbt -v test
+#      Uncomment after merge to master
```
Comment (adamw) [inline .github/workflows/ci.yml:35]: I'm not sure if this will work here as there are multiple independent projects

File: build.sbt (hunk 1 above)
Comment (Pask423) [reply build.sbt]: Yeah result of copy-pasting i fix comments after merge this line should be delete and we will have working mima for all projects

File: build.sbt [hunk 4]
```diff
@@ -18,7 +18,8 @@ lazy val commonSettings = commonSmlBuildSettings ++ ossPublishSettings ++ Seq(
   parallelExecution := false,
   // Sonatype OSS deployment
   credentials += Credentials(Path.userHome / ".ivy2" / ".credentials"),
-  Test / publishArtifact := false
+  Test / publishArtifact := false,
+  mimaPreviousArtifacts := Set.empty // remove after merge
```
Comment (adamw) [inline build.sbt:22]: adding mima in that way is pointless ;)

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+      - name: Check MiMa # disable for major releases
+        run: sbt -v mimaReportBinaryIssues
+      - name: Cleanup
+        run: |
+          rm -rf "$HOME/.ivy2/local" || true
+          find $HOME/.ivy2/cache                       -name "ivydata-*.properties" -delete || true
+          find $HOME/.ivy2/cache                       -name "*-LM-SNAPSHOT*"       -delete || true
+          find $HOME/.cache/coursier/v1                -name "ivydata-*.properties" -delete || true
+          find $HOME/.sbt                              -name "*.lock"               -delete || true
+  publish:
```
Comment (adamw) [inline .github/workflows/ci.yml:74]: can we publish this project using GHA? I don't think so, every artifact is versioned differently

**Comment by adamw** [comment]
Thanks :)

**Pull request merged**